catalog/async/catalog-async-processingframework/src/main/java/org/codice/ddf/catalog/async/processingframework/impl/InMemoryProcessingFramework.java

```diff
@@ -0,0 +1,326 @@
+                ProcessRequest<ProcessDeleteItem> request = input;
+
+                for (PostProcessPlugin plugin : postProcessPlugins) {
+                    try {
+                        request = plugin.processDelete(request);
+                    } catch (PluginExecutionException e) {
+                        LOGGER.debug("Unable to process request through plugin: {}",
+                                plugin.getClass()
+                                        .getCanonicalName(),
+                                e);
+                    }
+                }
+            });
+        }
+    }
+
+    private <T extends ProcessResourceItem> void closeInputStream(ProcessRequest<T> request) {
+        request.getProcessItems()
+                .stream()
+                .map(ProcessResourceItem::getProcessResource)
+                .filter(Objects::nonNull)
+                .forEach(resource -> {
+                    try {
+                        resource.getInputStream()
+                                .close();
+                    } catch (IOException e) {
+                        LOGGER.debug(
+                                "Failed to close stream of asynchronous request. There may be temporary files left in data/tmp.");
+                    }
+                });
+    }
+
+    private <T extends ProcessResourceItem> void storeProcessRequest(
+            ProcessRequest<T> processRequest) {
+        LOGGER.trace("Storing update request post processing change(s)");
+
+        Set<ContentItem> contentItemsToUpdate = new HashSet<>();
+        Set<Metacard> metacardsToUpdate = new HashSet<>();
+
+        List<TemporaryFileBackedOutputStream> tfbosToCleanUp = new ArrayList<>();
+        for (T item : processRequest.getProcessItems()) {
+            if (item.getProcessResource() == null && item.isMetacardModified()) {
+                metacardsToUpdate.add(item.getMetacard());
+            }
+
+            final ProcessResource processResource = item.getProcessResource();
+            if (processResource != null && processResource.isModified()) {
+                try {
+                    TemporaryFileBackedOutputStream tfbos = new TemporaryFileBackedOutputStream();
+                    long c = IOUtils.copyLarge(processResource.getInputStream(), tfbos);
+                    LOGGER.debug("Coppied {} bytes to TemporaryFileBackedOutputStream.", c);
+                    ByteSource byteSource = tfbos.asByteSource();
+
+                    ContentItem contentItem = new ContentItemImpl(item.getMetacard()
+                            .getId(),
+                            processResource.getQualifier(),
+                            byteSource,
+                            processResource.getMimeType(),
+                            processResource.getName(),
+                            processResource.getSize(),
+                            item.getMetacard());
+
+                    contentItemsToUpdate.add(contentItem);
```

Comment:
❗️ Before we create the FBOS and do all the above work, shouldn't we first make sure the `ContentItem` isn't already in the set? Otherwise I'm pretty sure we could end up creating FBOS that are not going to be used.